server/issue.go

```diff
@@ -20,6 +20,56 @@ import (
 	"github.com/mattermost/mattermost-plugin-jira/server/utils"
 )
 
+func makePost(userId, channelId, message string) *model.Post {
+	return &model.Post{
+		UserId:    userId,
+		ChannelId: channelId,
+		Message:   message,
+	}
+}
+
+func httpAPITransitionIssue(ji Instance, w http.ResponseWriter, r *http.Request) (int, error) {
+	requestData := model.PostActionIntegrationRequestFromJson(r.Body)
+
+	mattermostUserId := requestData.UserId
+	if mattermostUserId == "" {
+		return http.StatusUnauthorized, errors.New("not authorized")
+	}
+
+	issueKey := requestData.Context["issueKey"].(string)
```

Comment:
If the request is malformed, and `issueKey` is not present, I believe this line will panic. Perhaps we can use a check like `issueKey, ok := ...` to do some error checking. We can expect it is present in general, but we should be as panic-free as possible.